src/pydata_sphinx_theme/translator.py

```diff
@@ -98,18 +98,28 @@ def visit_reference(self, node):
         This will modify the ``href`` attribute of any internal HTML ``<a>`` tags, e.g.
         the sidebar navigation links.
         """
-        try:
-            # We are only interested in internal anchor references
-            internal, anchorname = node["internal"], node["anchorname"]
-            if internal and anchorname.startswith("#") and "." in anchorname:
+        conditions = (
+            ("refuri", node.get("internal", False)),
+            ("refid", True),
+            ("refuri", True),
+        )
+        for key, _internal in conditions:
+            refid = node.get(key, "")
+            if _internal and "." in refid:
```

Comment:
whatever the value of `node.get("internal", False)` the refuri key will be tested as True in l108 is it normal ? If yes why not simply use a list: ["refid", "refuri"] ?